lib/puppet/provider/tacacs_global/nxapi.rb

```diff
@@ -0,0 +1,136 @@
+# The NXAPI provider for tacacs_global.
+#
+# October, 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:tacacs_global).provide(:nxapi) do
+  desc 'The Cisco NXAPI provider for tacacs_global.'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  # The following properties are only valid when 'enable' is set to true
+  ENABLED_ONLY_PROPS = [:timeout, :key, :key_format]
+
+  def initialize(value={})
+    super(value)
+    @tacacs_global = Cisco::TacacsServer.new(false)
+    @property_flush = {}
+    debug 'Created provider instance of tacacs_global'
+  end
+
+  def self.get_properties(name)
+    debug "Checking instance, TacacsServer #{name}"
+
+    v = Cisco::TacacsServer.new(false)
+
+    current_state = {}
+    current_state[:name] = name
+    current_state[:enable] = Cisco::TacacsServer.enabled ? :true : :false
+
+    if current_state[:enable].eql?(:true)
+      current_state[:timeout] = v.timeout
+      current_state[:key] = v.encryption_password
+      current_state[:key_format] = v.encryption_type
```

Comment:
the value returned here is something like Cisco::TACACS_SERVER_ENC_CISCO_TYPE_7, I don't see where this is getting munged into a puppet-like value